### PR TITLE
[SPARK-4504][Examples] fix run-example failure if multiple example jars present

### PR DESCRIPTION
**Pull request by gvramana**
Fix run-examples script error during multiple jars.
Modified to support failing fast similar to assembly.jar

Author: Venkata Ramana G ramana.gollamudi@huawei.com

### CODE REVIEW

**Comment by srowen** [comment]
There was already a PR for this: https://github.com/apache/spark/pull/3069
But it seems to be fixing a different root cause, that the assembly generated by SBT and Maven are named differently. I am not sure if this also addresses this problem?

**Comment by JoshRosen** [comment]
@srowen You can still end up with multiple examples assemblies if you're switching between branches without performing cleans, e.g. switch from branch-1.2 to master.

**Comment by srowen** [comment]
@JoshRosen Yeah I get that; I think there is another different issue lurking here, though I haven't really looked at it, and that is that the glob pattern does not match what SBT builds. The other PR might also be worth merging.

**Comment by gvramana** [comment]
@nchammas @JoshRosen Thanks for your comments. I have updated the same. please review.

**Comment by nchammas** [comment]
Updated Bash style looks good to me.

**Comment by JoshRosen** [comment]
Jenkins, this is ok to test.

**Comment by JoshRosen** [comment]
LGTM.  Thanks @nchammas for looking this over.  I'm going to merge this into `master` (1.3.0), `branch-1.2` (1.2.1), and `branch-1.1` (1.1.2).

File: bin/compute-classpath.sh [hunk 1]
```diff
@@ -72,22 +72,25 @@ else
...
 
-num_jars="$(ls "$assembly_folder" | grep "spark-assembly.*hadoop.*\.jar$" | wc -l)"
-if [ "$num_jars" -eq "0" ]; then
-  echo "Failed to find Spark assembly in $assembly_folder"
-  echo "You need to build Spark before running this program."
-  exit 1
-fi
+num_jars=0
+ 
+for f in ${assembly_folder}/spark-assembly*hadoop*.jar; do
```
Comment (nchammas) [inline bin/compute-classpath.sh]: There is a word splitting bug here. My bad for missing this during review.

File: bin/compute-classpath.sh [hunk 2]
```diff
@@ -72,22 +72,25 @@ else
...
+  num_jars=$((num_jars+1))   
+done
+
 if [ "$num_jars" -gt "1" ]; then
-  jars_list=$(ls "$assembly_folder" | grep "spark-assembly.*hadoop.*.jar$")
-  echo "Found multiple Spark assembly jars in $assembly_folder:"
-  echo "$jars_list"
-  echo "Please remove all but one jar."
+  echo "Found multiple Spark assembly jars in $assembly_folder:" 1>&2
+  ls ${assembly_folder}/spark-assembly*hadoop*.jar 1>&2
```
Comment (nchammas) [inline bin/compute-classpath.sh]: There is another word splitting bug here. My bad for missing it.

File: bin/run-example [hunk 3]
```diff
@@ -35,17 +35,32 @@ else
...
+else
+  JAR_PATH="${EXAMPLES_DIR}/target/scala-${SPARK_SCALA_VERSION}"
 fi
 
-if [[ -z "$SPARK_EXAMPLES_JAR" ]]; then
-  echo "Failed to find Spark examples assembly in $FWDIR/lib or $FWDIR/examples/target" 1>&2
-  echo "You need to build Spark before running this program" 1>&2
+JAR_COUNT=0
+
+for f in ${JAR_PATH}/spark-examples-*hadoop*.jar; do
```
Comment (nchammas) [inline bin/run-example]: Same: Word splitting

File: bin/run-example [hunk 4]
```diff
@@ -35,17 +35,32 @@ else
...
+    echo "You need to build Spark before running this program" 1>&2
+    exit 1
+  fi 
+  SPARK_EXAMPLES_JAR="$f"
+  JAR_COUNT=$((JAR_COUNT+1))   
+done
+
+if [ "$JAR_COUNT" -gt "1" ]; then
+  echo "Found multiple Spark examples assembly jars in ${JAR_PATH}" 1>&2
+  ls ${JAR_PATH}/spark-examples-*hadoop*.jar 1>&2
```
Comment (nchammas) [inline bin/run-example]: Same: Word splitting

**Comment by gvramana** [comment]
Oh, basic mistake :( . Sorry I missed it completely. 
Fixed the same in PR #4561
@nchammas Thanks for identifying.

**Pull request closed without merging**